addons/product/models/product.py

```diff
@@ -658,7 +658,7 @@ def _get_filtered_sellers(self, partner_id=False, quantity=0.0, date=None, uom_i
         precision = self.env['decimal.precision'].precision_get('Product Unit of Measure')
 
         sellers_filtered = self._prepare_sellers(params)
-        sellers_filtered = sellers_filtered.filtered(lambda s: not s.company_id or s.company_id.id == self.env.company.id)
+        sellers_filtered = sellers_filtered.filtered(lambda s: not s.company_id or s.company_id.id in self.env.companies.ids)
```

Comment:
I realized that if you have a supplierinfo for both companies, it will always take the first instead of the one(s) that correspond to the user's default company
```suggestion
                if self.env.company in sellers_filtered.company_id:
                    sellers_filtered = sellers_filtered.filtered(lambda s: not s.company_id or s.company_id.id == self.env.company.id)
                else:
                    sellers_filtered = sellers_filtered.filtered(lambda s: not s.company_id or s.company_id.id in self.env.companies.ids)
```

Reply:
Thank you for your deepening, I added your changes.